### ISSUE TITLE
Pino v7 trasnports + AWS lambda

### ISSUE THREAD

**Issue by royhamulak** [pinojs/pino#1217]
When trying to log on AWS lambda with the new trasnports api logs are being lost.

I'm using multiple transports for the logger, one of them being `pino/file` which is responsible for the `stdout` logs

this is the gist of my configuration:
```javascript
const transports = pino.trasnports({
  targets: [
    {
      target: 'pino/file',
      options: {},
      level,
    },
    {
      target: 'my-other-transport',
      options: {},
      level,
    },
  ],
})

pino({...my_options}, transports)

```

when calling `logger.info('blah')` with the above setup, it seems that nothing registers to the stdout of the lambda.

I looked at some of the documentation and I found out that [when using async logging mode, you should use the `flushSync()` method](https://github.com/pinojs/pino/blob/master/docs/asynchronous.md#aws-lambda) after every function call in order to avoid losing data in AWS lambdas.

My current workaround seems something like this:
```javascript
pino({
  ...my_options,
  hooks: {
    logMethod: function (inputArgs, method, level) {
      const res = method.apply(this, inputArgs)
      this[pino.symbols.streamSym].flushSync()
      return res
    }
  }
}, transports)
```

Can anything else be done here? Maybe some sort of `autoDrain` option?

**Comment by simoneb** [comment]
Thanks for reporting this. As a temporary workaround you may want to consider looking into https://github.com/FormidableLabs/pino-lambda, which is a drop in replacement for pino which works in a lambda environment.

**Comment by mcollina** [comment]
@simoneb I think we should be creating an entry in our FAQ for this. Would you like to send a PR?

**Comment by jsumners** [comment]
I don't think we should rely on a third party being documented as a solution to the problem. Adding them to the FAQ or something as a thing that exists is fine. But we should document the above "hack" or implement a direct feature for this, as we did in v6.

**Comment by simoneb** [comment]
I agree James. Also note that I'd also proposed the folks of that library to turn it into something which can be plugged into pino, rather than replace pino. Whole conversation here https://github.com/FormidableLabs/pino-lambda/issues/24

**Comment by mcollina** [comment]
> I don't think we should rely on a third party being documented as a solution to the problem. Adding them to the FAQ or something as a thing that exists is fine. But we should document the above "hack" or implement a direct feature for this, as we did in v6.

I would not do the `flushSync()` after every log line. It completely defeats the purpose of having transports to begin with (`flushSync()` blocks the main thread). You'd need to do it once that at the end of your function.

**Comment by royhamulak** [comment]
> I would not do the flushSync() after every log line. It completely defeats the purpose of having transports to begin with (flushSync() blocks the main thread). You'd need to do it once that at the end of your function.

One of my transports is aggregating the messages and then sends it to a 3rd party (logz.io in this particular case).

As I tried different things I encountered an issue where there was no normal way to notify the ThreadStream that the stream is still processing (even with delaying the end() call to after the request finished sending), thus terminating the lambda.

We moved from this solution due to other factors (changed the way we gather logs) but the issue still exits, I'll try to reproduce the above problem and open a ticket

**Comment by mcollina** [comment]
I think we should link to that module in our docs!

**Comment by simoneb** [comment]
I believe this issue can be closed. The solution is to use pino-lambda plugin
